codemod_yaml/items.py

```diff
@@ -649,7 +660,7 @@ def __getitem__(self, key: Any) -> Item:
 
     def __setitem__(self, key: Any, value: Any) -> None:
         key = item(key)
-        pair: Optional[MappingPair] = self.get(key, None)  # type: ignore[assignment]
+        pair: Optional[MappingPair] = dict.get(self, key, None)  # type: ignore[misc]
```

Comment:
This is slightly mind-bendy. You're calling the builtin dict.get() and using it like a method on a class? 🤯

Reply:
The `__new__` means we _are_ a dict.  The `type: ignore` is another mind-bendy part, we lie about the type that gets stored in the dict, because that's how our users will see it -- but we're actually storing pairs, not values.